analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -34,36 +34,62 @@ public abstract class TokenTypeAnalyzerBase<TSyntaxKind> : UtilityAnalyzerBase<T
 
         protected TokenTypeAnalyzerBase() : base(DiagnosticId, Title) { }
 
-        protected abstract TokenClassifierBase GetTokenClassifier(SyntaxToken token, SemanticModel semanticModel, bool skipIdentifierTokens);
+        protected abstract TokenClassifierBase GetTokenClassifier(SemanticModel semanticModel, bool skipIdentifierTokens);
+        protected abstract TriviaClassifierBase GetTriviaClassifier();
```

Comment:
I don't have a good answer to that. It looks indeed a bit over-engineered and you are right about merging the classes. But let's keep it for the moment as the next step is to avoid semantic model lookups as much as possible and to do so, the classes will grow.
As is, the class split makes sense (have different classes for orchestration and token/trivia handling), and putting everything in a single class might violate the "class complexity too high" rule at one point.